components/script/dom/htmlinputelement.rs

```diff
@@ -410,8 +410,19 @@ impl HTMLInputElementMethods for HTMLInputElement {
     fn SetValue(&self, value: DOMString) -> ErrorResult {
         match self.value_mode() {
             ValueMode::Value => {
+                // Step 1.
+                let old_value = self.textinput.borrow().get_content();
+                // Step 2.
                 self.textinput.borrow_mut().set_content(value);
+                // Step 3.
                 self.value_dirty.set(true);
+                // Step 4.
+                self.sanitize_value();
+                // Step 5.
+                if self.textinput.borrow().get_content() != old_value {
```

Comment:
This clones the new value for no reason.

Reply:
Which method call is a clone?

Reply:
get_content.

Reply:
Then I'm not sure which other method gets the content of the text input without using `get_content`.